checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,363 @@
+    return type;
+  }
+
+  /**
+   * Creates a {@link MustCall} annotation whose values are the given strings.
+   *
+   * @param val the methods that should be called
+   * @return an annotation indicating that the given methods should be called
+   */
+  public AnnotationMirror createMustCall(final String... val) {
+    AnnotationBuilder builder = new AnnotationBuilder(processingEnv, MustCall.class);
+    Arrays.sort(val);
+    builder.setValue("value", val);
+    return builder.build();
+  }
+
+  @Override
+  public QualifierHierarchy createQualifierHierarchy() {
+    return new SubtypeIsSubsetQualifierHierarchy(
+        this.getSupportedTypeQualifiers(), this.getProcessingEnv());
+  }
+
+  /**
+   * Fetches the store from the results of dataflow for block. If useBlock is true, then the store
+   * after block is returned; if useBlock is false, the store before succ is returned.
+   *
+   * @param useBlock whether to use the store after the block itself or the store before its
+   *     successor, succ
+   * @param block a block
+   * @param succ block's successor
+   * @return the appropriate CFStore, populated with MustCall annotations, from the results of
+   *     running dataflow
+   */
+  public CFStore getStoreForBlock(boolean useBlock, Block block, Block succ) {
+    return useBlock ? flowResult.getStoreAfter(block) : flowResult.getStoreBefore(succ);
+  }
+
+  /**
+   * Get the element for a single CreatesObligation annotation.
+   *
+   * @return the element
+   */
+  @Override
+  public ExecutableElement getCreatesObligationValueElement() {
+    return createsObligationValueElement;
+  }
+
+  /**
+   * Get the element for a list CreatesObligation annotation.
+   *
+   * @return the element
+   */
+  @Override
+  public ExecutableElement getCreatesObligationListValueElement() {
+    return createsObligationListValueElement;
+  }
+
+  /**
+   * The TreeAnnotator for the MustCall type system. This tree annotator treats non-owning method
+   * parameters as bottom, regardless of their declared type, when they appear in the body of the
+   * method. Doing so is safe because being non-owning means, by definition, that their must-call
+   * obligations are only relevant at the call site.
+   */
+  private class MustCallTreeAnnotator extends TreeAnnotator {
+    /**
+     * Create a MustCallTreeAnnotator
+     *
+     * @param mustCallAnnotatedTypeFactory the type factory
+     */
+    public MustCallTreeAnnotator(MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory) {
+      super(mustCallAnnotatedTypeFactory);
+    }
+
+    // When they appear in the body of a method or constructor, treat non-owning parameters
+    // as bottom regardless of their declared type.
+    @Override
+    public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror type) {
+      Element elt = TreeUtils.elementFromTree(node);
+      if (elt.getKind() == ElementKind.PARAMETER
+          && (checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
```

Comment:
The Javadoc does not mention this aspect of the behavior.

Also, I suggest creating a boolean field in this class.  Then it can be queried cheaply.

Reply:
Can you respond to [my comment](https://github.com/typetools/checker-framework/pull/4484#discussion_r610949660) about `hasOption` from the previous round of reviews?